hadoop/src/main/java/org/apache/carbondata/hadoop/streaming/CarbonStreamRecordWriter.java

```diff
@@ -99,13 +100,21 @@
     initialize(job);
   }
 
+  public CarbonStreamRecordWriter(TaskAttemptContext job, CarbonLoadModel carbonLoadModel)
+      throws IOException {
+    this.carbonLoadModel = carbonLoadModel;
+    initialize(job);
+  }
+
   private void initialize(TaskAttemptContext job) throws IOException {
     // set basic information
     hadoopConf = job.getConfiguration();
-    CarbonLoadModel carbonLoadModel = CarbonStreamOutputFormat.getCarbonLoadModel(hadoopConf);
     if (carbonLoadModel == null) {
-      throw new IOException(
-          "CarbonStreamRecordWriter require configuration: mapreduce.output.carbon.load.model");
+      carbonLoadModel = CarbonStreamOutputFormat.getCarbonLoadModel(hadoopConf);
+      if (carbonLoadModel == null) {
+        throw new IOException(
+            "CarbonStreamRecordWriter require configuration: mapreduce.output.carbon.load.model");
```

Comment:
This configuration should start with `carbon`, change to `carbon.load.model`

Reply:
Do you really require the whole CarbonLoadModel? It seems you only need CarbonTable object, right?